### PR TITLE
multistage docker build

### PR DESCRIPTION
**Pull request by jaydouble**
I've made a multistage dockerfile.
1 for building the gateway, and 1 for running.

The running one is now only 8Mb. 
https://hub.docker.com/r/jaydouble/rdpgw

### CODE REVIEW

File: dev/docker/Dockerfile [hunk 1]
```diff
@@ -1,47 +1,49 @@
...
-    mkdir -p /opt/rdpgw && \
-    mv /rdpgw/bin/rdpgw /opt/rdpgw/rdpgw && \
-    rm -rf /root/go && \
-    rm -rf /rdpgw
-
-RUN rm -rf /go
+# add user
+RUN adduser --disabled-password --gecos "" --home /opt/rdpgw --uid 1001 rdgw
 
+# build rdwgw and set rights
```
Comment (bolkedebruin) [inline dev/docker/Dockerfile:27]: nit: `rdpgw` not `rdwgw`

File: dev/docker/Dockerfile [hunk 2]
```diff
@@ -1,47 +1,49 @@
...
-    go mod tidy && \
-    go build -trimpath -tags '' -ldflags '' -o '/rdpgw/bin/rdpgw' ./cmd/rdpgw && \
-    mkdir -p /opt/rdpgw && \
-    mv /rdpgw/bin/rdpgw /opt/rdpgw/rdpgw && \
-    rm -rf /root/go && \
-    rm -rf /rdpgw
-
-RUN rm -rf /go
+# add user
+RUN adduser --disabled-password --gecos "" --home /opt/rdpgw --uid 1001 rdgw
```
Comment (bolkedebruin) [inline dev/docker/Dockerfile:25]: it night be smart to use `rdpgw` as a user as it is in line with the project name `rdgw` might create confusion

File: dev/docker/Dockerfile [hunk 3]
```diff
@@ -1,47 +1,49 @@
...
-RUN adduser --disabled-password --gecos "" -h /opt/rdpgw -u 1001 rdgw && \
-    mv server.pem /opt/rdpgw/server.pem && \
-    mv key.pem /opt/rdpgw/key.pem && \
-    chown -R 1001 /opt/rdpgw && \
-    chmod +x /opt/rdpgw/rdpgw
-
 USER 1001
 WORKDIR /opt/rdpgw
-ENTRYPOINT /opt/rdpgw/rdpgw
+ENTRYPOINT /opt/rdpgw/rdpgw
```
[the remark this answers is not in the document]
Comment (jaydouble) [reply dev/docker/Dockerfile]: What do you exactly mean with fix the newline?

File: dev/docker/Dockerfile (hunk 1 above)
Comment (jaydouble) [reply dev/docker/Dockerfile]: fixed (sorry, I'm a little dyslexic)

File: dev/docker/Dockerfile (hunk 2 above)
Comment (jaydouble) [reply dev/docker/Dockerfile]: fixed (sorry, I'm a little dyslexic)

**Pull request merged**